### ISSUE TITLE
Image.Update.Post [stashdb performer gallery] Python was not found; run without arguments to install from the Microsoft Store

### ISSUE THREAD

**Issue by mbh9700** [stashapp/CommunityScripts#468]
I am getting tons of these errors when scanning my library.
Immediately after the Python was not found error, it also gives this error:
Image.Update.Post [stashdb performer gallery]: returned error: exit status 9009
And also:
error generating sprite: ffmpeg command produced no output: <-v error -y -ss 20.23762962962963 -i D:\xvids\30076_2.wmv -frames:v 1 -vf scale=160:-2 -c:v bmp -f rawvideo ->

Python is in the path. If I open a command prompt in the folder that stash is installed, and type Python, I get the python prompt.

So, what is having a problem finding Python, and how do I fix it? Are the other errors related?

**Comment by Maista6969** [comment]
There are many reasons that Stash could have trouble finding the correct Python installation: you might have Python added to your user path, for example, but you're running Stash as an Administrator. There are also issues around permissions when Python is installed to your %USERPROFILE%, which is why e.g. the [CommunityScrapers README](https://github.com/stashapp/CommunityScrapers?tab=readme-ov-file#python-scrapers) recommends installing it somewhere like `C:\Python` instead.

Either way, you can explicitly tell Stash which Python it should use in `Settings > System > Application paths`

**Comment by mbh9700** [comment]
Ok, I set the "Python Executable Path" to point directly to the python.exe file, and now all of the files are getting this sequence:

INFO[2024-11-24 15:59:52] rescanning D:\xvids\3466_3.wmv
INFO[2024-11-24 15:59:52] Calculating fingerprints for D:\xvids\3466_3.wmv ...
ERRO[2024-11-24 15:59:52] [Plugin / Tag Scenes From Performer Tags] Traceback (most recent call last):
ERRO[2024-11-24 15:59:52] [Plugin / Tag Scenes From Performer Tags]   File "G:\x\organizers\Stash\plugins\community\tagScenesWithPerfTags\tagScenesWithPerfTags.py", line 1, in <module>
ERRO[2024-11-24 15:59:52] [Plugin / Tag Scenes From Performer Tags]     from stashapi.stashapp import StashInterface
ERRO[2024-11-24 15:59:52] [Plugin / Tag Scenes From Performer Tags] ModuleNotFoundError: No module named 'stashapi'
ERRO[2024-11-24 15:59:52] Scene.Update.Post [Tag Scenes From Performer Tags]: returned error: exit status 1

**Comment by Maista6969** [comment]
Yes, now you will need to install the Python dependencies: there are some scrapers and plugins that can do this for themselves, but most of them are old enough that they rely on the user to handle this. See the link I posted previously for a copy-pastable command that will install the correct dependencies for 99% of Stash scrapers and plugins

**Comment by mbh9700** [comment]
Ok, followed the instructions.
* Exited stash
* Download get-pip.py
* Upgraded Python from 3.12.1 to 3.12.2
* Installed get-pip.py
* pip install command for the dependencies

Getting much less errors. I am still seeing this one sometimes:
INFO[2024-11-24 16:33:23] Calculating fingerprints for D:\xvids\0214786.flv ...
ERRO[2024-11-24 16:33:24] error generating sprite: ffmpeg command produced no output: <-v error -y -ss 224.55395061728393 -i D:\xvids\0208725.flv -frames:v 1 -vf scale=160:-2 -c:v bmp -f rawvideo ->

**Comment by mbh9700** [comment]
So it's still scanning lol. Big library I guess. Or slow CPU.
Very rarely getting an error.
Still getting those ffmpeg ones above.

Thanks so much Maista6969!
